src/android/NetworkManager.java

```diff
@@ -230,9 +230,44 @@ private void updateConnectionInfo(NetworkInfo info) {
 
             sendUpdate(connectionType);
             lastInfo = thisInfo;
+        } else {
+            LOG.d(LOG_TAG, "Networkinfo state didn't change, there is no event propagated to the javascript side.");
         }
     }
 
+    private boolean connectionInfoDiffersFromLastInfo(JSONObject thisInfo) {
+        // JSONObject.equals does not work, so the equals is done explicit for every key in the object
+        if(lastInfo == null) {
+            return thisInfo != null;
+        }
+
+        if(thisInfo == null) {
+            return true;
+        }
+
+        if (valueInJSONObjectDiffersForKey(thisInfo , lastInfo, "type")) {
+            return true;
+        }
+
+        return valueInJSONObjectDiffersForKey(thisInfo , lastInfo, "extraInfo");
+    }
+
+    private boolean valueInJSONObjectDiffersForKey(JSONObject firstObject, JSONObject secondObject, String key) {
```

Comment:
Maybe we should just get rid of the JSONObject in this plugin.
We could refactor it to a seperate (VO-class) with two fields: 'type' and 'extraInfo'.
This class could have a correct equals and hashcode function.

This could make the try - catch constructions with JSONException obsolete on a lof of places.

Reply:
As long as this doesn't affect the `PluginResult` responses, then I don't have any objection. But do you think it's necessary to be as a part of a bugfix PR, or should that change be a separate refactor PR?

Reply:
I did take a second look to the code.
The pluginResult is just the "type" of network.

So this made my reflect a little bit about the "extraInfo" state.

```
JSONObject connectionInfo = new JSONObject();
 try {
       connectionInfo.put("type", type);
       connectionInfo.put("extraInfo", extraInfo);
  } catch (JSONException e) {
      LOG.d(LOG_TAG, e.getLocalizedMessage());
 }
```

If I read the code the "extraInfo" is not used at all, it is stored in the JsonObject, but is not used or read anywhere else in the NetworkManager class.

This remark is just about the boyscout rule ( Leave your code better than you found it ).
Maybe a next developer is happy that someone left it cleaner. 
So my suggestions are: 
- remove the JSONObject, just use a String for type as state.
- remove the "extraInfo" state.

I leave it over to your decision if it should be a seperate PR, or not.
But I am willing to make these changes in this or another PR. (It's milestone 3, so there is time?)

Reply:
Done.
I am not 100% with the naming of the method: getTypeOfNetworkFallbackToTypeNoneIfNotConnected , but it is okay.

My tests on the emulator were successfull. I am planning on doing some tests on my mobile phone.

There is still more unused code in the plugin in my opinion, but that's for another iteration.

Reply:
Tests on the mobile phone were successfull. Seems to work.